homeassistant/components/goodwe/__init__.py

```diff
@@ -19,19 +19,21 @@
 from .coordinator import GoodweUpdateCoordinator
 
 
+async def _connect_inverter(entry: ConfigEntry) -> Inverter:
```

Comment:
Techniqually the move of the _connect_inverter is not part of adding diagnostic info.
It is not related right?

Reply:
It actually is. In the test, I needed to mock the integration setup (connection inverter) and wasn't able to do it without extracting to separate function. (Mocking the goodwe library call was not working, or at least I wasn't able to make it work.).

Reply:
We shouldn't do that.
We should patch the upstream library instead of creating a mock of the inverter and patch internals.

Reply:
I tend to agree, but as I mentioned above, I have tried but failed, patching the library somehow did not work. I'm afraid I need help, since I can't do it.

Reply:
You should patch the `connect` callable instead and revert the changes in this file.
See my other comment